packages/babel-cli/src/babel/dir.js

```diff
@@ -13,10 +13,19 @@ module.exports = function (commander, filenames) {
 
     let dest = path.join(commander.outDir, relative);
 
-    let data = util.compile(src, {
-      sourceFileName: slash(path.relative(dest + "/..", src)),
-      sourceMapTarget: path.basename(relative)
-    });
+    let sourceFilePath = path.relative(dest + "/..", src);
+    let sourceMapPath = src + ".map";
```

Comment:
We shouldn't make assumptions about filename mappings. If we're doing to use input sourcemaps, we should read the file-end comment like everything else. I also think enabling this behavior would be a breaking change, since you never know what workflows people have in place, so we'd have to make an option for it.